Tests/SkillFunctionalTests/FileUpload/FileUploadTests.cs

```diff
@@ -0,0 +1,115 @@
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using SkillFunctionalTests.Common;
+using TranscriptTestRunner;
+using TranscriptTestRunner.XUnit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SkillFunctionalTests.FileUpload
+{
+    [Trait("TestCategory", "FileUpload")]
+    public class FileUploadTests : ScriptTestBase
+    {
+        private readonly string _testScriptsFolder = Directory.GetCurrentDirectory() + @"/FileUpload/TestScripts";
+
+        public FileUploadTests(ITestOutputHelper output)
+            : base(output)
+        {
+        }
+
+        public static IEnumerable<object[]> TestCases()
+        {
+            var channelIds = new List<string> { Channels.Directline };
+
+            var deliverModes = new List<string>
+            {
+                DeliveryModes.Normal,
+                DeliveryModes.ExpectReplies
+            };
+
+            var hostBots = new List<HostBot>
+            {
+                HostBot.WaterfallHostBotDotNet,
+
+                // TODO: Enable these when the ports to JS, Python and composer are ready
+                //HostBotNames.WaterfallHostBotJS,
+                //HostBotNames.WaterfallHostBotPython,
+                //HostBotNames.ComposerHostBotDotNet
+            };
+
+            var targetSkills = new List<string>
+            {
+                SkillBotNames.WaterfallSkillBotDotNet,
+
+                // TODO: Enable these when the ports to JS, Python and composer are ready
+                //SkillBotNames.WaterfallSkillBotJS,
+                //SkillBotNames.WaterfallSkillBotPython,
+                //SkillBotNames.ComposerSkillBotDotNet
+            };
+
+            var scripts = new List<string>
+            {
+                "FileUpload1.json",
+            };
+
+            var testCaseBuilder = new TestCaseBuilder();
+            var testCases = testCaseBuilder.BuildTestCases(channelIds, deliverModes, hostBots, targetSkills, scripts);
+            foreach (var testCase in testCases)
+            {
+                yield return testCase;
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(TestCases))]
+        public async Task RunTestCases(TestCaseDataObject testData)
+        {
+            var testCase = testData.GetObject<TestCase>();
+            Logger.LogInformation(JsonConvert.SerializeObject(testCase, Formatting.Indented));
+
+            var options = TestClientOptions[testCase.HostBot];
+            var runner = new XUnitTestRunner(new TestClientFactory(testCase.ChannelId, options, Logger).GetTestClient(), TestRequestTimeout, Logger);
+
+            // Execute the first part of the conversation.
+            var testParams = new Dictionary<string, string>
+            {
+                { "DeliveryMode", testCase.DeliveryMode },
+                { "TargetSkill", testCase.TargetSkill },
+                { "FileLocation", "\\\"C:\\\\local\\\\Temp\\\\architecture-resize.png\\\"" }
```

Comment:
Hi @gabog, this is the path to the temp folder where the bot (in Azure) saves the file. We can add a comment to clarify this allowing the user to change this path when running the bots locally, or just removed the assertion of this part of the message.

Reply:
Ah, OK, see my comment in the last review, Python bots will probably use a different path in linux and this may need to get parametrized, we can address this once we port to python and solve the other issue I raised.